prometheus_client/exposition.py

```diff
@@ -271,27 +272,34 @@ def factory(cls, registry):
         return MyMetricsHandler
 
 
-def write_to_textfile(path, registry):
+def write_to_textfile(path: str, registry: CollectorRegistry) -> None:
     """Write metrics to the given path.
 
     This is intended for use with the Node exporter textfile collector.
     The path must end in .prom for the textfile collector to process it."""
     tmppath = f'{path}.{os.getpid()}.{threading.current_thread().ident}'
     with open(tmppath, 'wb') as f:
         f.write(generate_latest(registry))
-    
+
     # rename(2) is atomic but fails on Windows if the destination file exists
     if os.name == 'nt':
         os.replace(tmppath, path)
     else:
         os.rename(tmppath, path)
 
 
-def _make_handler(url, method, timeout, headers, data, base_handler):
+def _make_handler(
+    url: str,
+    method: str,
+    timeout: Optional[float],
```

Comment:
Seems fine to me, but most cases where I see `Optional`, there is typically a default of `=None`; but in this case I guess the caller of this is explicitly required to pass in an explicit `None`..

Reply:
Yeah, this one is an internal and requires that we explicitly pass timeout though even if the value is None (or a default value) from one of the other handlers. A bit unusual I agree.